packages/authentication/ios/Plugin/FirebaseAuthentication.swift

```diff
@@ -157,6 +167,27 @@ public typealias AuthStateChangedObserver = () -> Void
         }
     }
 
+    @objc func signInWithEmailLink(_ call: CAPPluginCall) {
+        let email = call.getString("email", "")
+        let emailLink = call.getString("emailLink", "")
+
+        if self.isSignInWithEmailLink(link: emailLink) {
```

Comment:
Either you have to handle the else case here and throw an exception or you pass the call to the Firebase SDK and it throws the exception. I suggest to just remove this if condition.